src/Fantomas/SourceParser.fs

```diff
@@ -1763,13 +1763,28 @@ let (|IndexWithoutDotExpr|ElmishReactWithoutChildren|ElmishReactWithChildren|Non
                    (SynExpr.ArrayOrListComputed (isArray = false; expr = indexExpr) as argExpr),
                    _) when (RangeHelpers.isAdjacentTo identifierExpr.Range argExpr.Range) ->
         IndexWithoutDotExpr(identifierExpr, indexExpr)
-    | SynExpr.App (_, false, OptVar (ident, _, _), ArrayOrList (sr, isArray, children, er, _), _) ->
-        ElmishReactWithoutChildren(ident, sr, isArray, children, er)
-    | SynExpr.App (_,
-                   false,
-                   SynExpr.App (_, false, OptVar ident, (ArrayOrList _ as attributes), _),
-                   ArrayOrList (sr, isArray, children, er, r),
-                   _) -> ElmishReactWithChildren(ident, attributes, (isArray, sr, children, er))
+        // ArrayOrList (sr, isArray, children, er, _)
+    // | SynExpr.App (_, false, OptVar (ident, _, _),  , _) ->
+
+    //     FunctionApplicationSingleList(ident, sr, isArray, children, er)
+    // | SynExpr.App (_,
+    //                false,
+    //                SynExpr.App (_, false, OptVar ident, (ArrayOrList _ as attributes), _),
+    //                ArrayOrList (sr, isArray, children, er, r),
+    //                _) -> FunctionApplicationDualList(ident, attributes, (isArray, sr, children, er))
+    | App(expr, exprs) ->
+        match exprs |> List.rev with
+
+        | (ArrayOrList (sr, isArray, children, er, _))::(ArrayOrList (sr2, isArray2, children2, er2, _))::tail ->  
```

Comment:
This about this, should we be hard coding single/dual array? Would it make sense to keep applying the formatting if the last args are all `ArrayOrList`? Such as

```fsharp
someFoo "This thing" [
  a
  b
] [
  a
  b
] [ 
  a
  b
]
```

Reply:
Perhaps if you captured all arrays/lists inside a single active pattern, similar to `FunctionApplicationSingleList`, `FunctionApplicationDualList` this idea might work.

Reply:
Yeah, just seeing how this is playing out it seems like a pattern that could emerge, but I'd also be fine with saying "two is the limit" at least for now.